### ISSUE TITLE
[Component] [tag] el-tag is not WCAG 2.1 AA compliant

### ISSUE THREAD

**Issue by MrBramard** [element-plus/element-plus#15495]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.4.11`
- Element Plus Version: `2.5.1`
- Browser / OS: `Chrome 120.0.6099.216 / Intel Mac OS 14.2.1`
- Build Tool: `Vite`

### Related Component

- `el-tag`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXRhZ1xuICAgIHYtZm9yPVwidGFnIGluIGR5bmFtaWNUYWdzXCJcbiAgICA6a2V5PVwidGFnXCJcbiAgICBjbGFzcz1cIm14LTFcIlxuICAgIGNsb3NhYmxlXG4gICAgOmRpc2FibGUtdHJhbnNpdGlvbnM9XCJmYWxzZVwiXG4gICAgQGNsb3NlPVwiaGFuZGxlQ2xvc2UodGFnKVwiXG4gID5cbiAgICB7eyB0YWcgfX1cbiAgPC9lbC10YWc+XG4gIDxlbC1pbnB1dFxuICAgIHYtaWY9XCJpbnB1dFZpc2libGVcIlxuICAgIHJlZj1cIklucHV0UmVmXCJcbiAgICB2LW1vZGVsPVwiaW5wdXRWYWx1ZVwiXG4gICAgY2xhc3M9XCJtbC0xIHctMjBcIlxuICAgIHNpemU9XCJzbWFsbFwiXG4gICAgQGtleXVwLmVudGVyPVwiaGFuZGxlSW5wdXRDb25maXJtXCJcbiAgICBAYmx1cj1cImhhbmRsZUlucHV0Q29uZmlybVwiXG4gIC8+XG4gIDxlbC1idXR0b24gdi1lbHNlIGNsYXNzPVwiYnV0dG9uLW5ldy10YWcgbWwtMVwiIHNpemU9XCJzbWFsbFwiIEBjbGljaz1cInNob3dJbnB1dFwiPlxuICAgICsgTmV3IFRhZ1xuICA8L2VsLWJ1dHRvbj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyBuZXh0VGljaywgcmVmIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgRWxJbnB1dCB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuY29uc3QgaW5wdXRWYWx1ZSA9IHJlZignJylcbmNvbnN0IGR5bmFtaWNUYWdzID0gcmVmKFsnVGFnIDEnLCAnVGFnIDInLCAnVGFnIDMnXSlcbmNvbnN0IGlucHV0VmlzaWJsZSA9IHJlZihmYWxzZSlcbmNvbnN0IElucHV0UmVmID0gcmVmPEluc3RhbmNlVHlwZTx0eXBlb2YgRWxJbnB1dD4+KClcblxuY29uc3QgaGFuZGxlQ2xvc2UgPSAodGFnOiBzdHJpbmcpID0+IHtcbiAgZHluYW1pY1RhZ3MudmFsdWUuc3BsaWNlKGR5bmFtaWNUYWdzLnZhbHVlLmluZGV4T2YodGFnKSwgMSlcbn1cblxuY29uc3Qgc2hvd0lucHV0ID0gKCkgPT4ge1xuICBpbnB1dFZpc2libGUudmFsdWUgPSB0cnVlXG4gIG5leHRUaWNrKCgpID0+IHtcbiAgICBJbnB1dFJlZi52YWx1ZSEuaW5wdXQhLmZvY3VzKClcbiAgfSlcbn1cblxuY29uc3QgaGFuZGxlSW5wdXRDb25maXJtID0gKCkgPT4ge1xuICBpZiAoaW5wdXRWYWx1ZS52YWx1ZSkge1xuICAgIGR5bmFtaWNUYWdzLnZhbHVlLnB1c2goaW5wdXRWYWx1ZS52YWx1ZSlcbiAgfVxuICBpbnB1dFZpc2libGUudmFsdWUgPSBmYWxzZVxuICBpbnB1dFZhbHVlLnZhbHVlID0gJydcbn1cbjwvc2NyaXB0PlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

### Steps to reproduce

Open your screen reader (VoiceOver for Mac)
Move though the tags using your screen reader

### What is Expected?

The screen reader should give the user all the information provided by the tag :
- User should know is entering and leaving a tag
- The close button should be read as a true button and not just an undefined image
- The user should be able to interact with the close button using there keyboard

### What is actually happening?

As you move through the tag using a screen reader, no information is given to the user.
The screen reader only reads the tag text without any context and reads "Image" for the close button

### Additional comments

A growing number of companies are required to comply to the level AA of the WCAG 2.1 to increase the accessibility of their websites

- The root div should have an aria-label telling the user that this component is a tag
- the image that acts as a close button should have a role="button"
- Any other information that we want to give to the user should be accessible to the screen reader using aria-label, aria-labelledby, aria-describedby, etc..
